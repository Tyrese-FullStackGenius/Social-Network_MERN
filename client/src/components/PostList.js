import React from 'react';
import PropTypes from 'prop-types';
import Post from './Post';

// const PostList = ({ posts }) => (
//   <div>
// {posts.map(post => (
//   <Post key={post._id} _id={post._id} text={post.text} />
// ))}
//   </div>
// );

class PostList extends React.Component {
  render() {
    console.log('postlist props', this.props);
    const { posts, deletePost } = this.props;
    return (
      <div>
        {posts.map(post => (
          <Post
            key={post._id}
            _id={post._id}
            text={post.text}
            deletePost={id => deletePost(id)}
          />
        ))}
      </div>
    );
  }
}

PostList.defaultProps = {
  posts: []
};

PostList.propTypes = {
  posts: PropTypes.arrayOf(
    PropTypes.shape({
      _id: PropTypes.string.isRequired,
      text: PropTypes.string.isRequired
    })
  ),
  deletePost: PropTypes.func.isRequired
};

export default PostList;
