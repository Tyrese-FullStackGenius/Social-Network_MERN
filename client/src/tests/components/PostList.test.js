import React from 'react';
import { shallow } from 'enzyme';
import PostList from '../../components/PostList';

it('should render PostList', () => {
  const wrapper = shallow(
    <PostList
      addComment={jest.fn()}
      deleteComment={jest.fn()}
      deletePost={jest.fn()}
      editComment={jest.fn()}
      editPost={jest.fn()}
      getPosts={jest.fn(() => Promise.resolve('posts'))}
      getFollowing={jest.fn(() => Promise.resolve('following'))}
      getUser={jest.fn()}
      posts={[
        {
          _id: '',
          author: '',
          authorId: '',
          likers: [],
          likesCount: 0,
          text: '',
          timestamp: 0
        }
      ]}
      updatePostLikes={jest.fn()}
      user={{ userId: 'abc123' }}
    />
  );
  expect(wrapper).toMatchSnapshot();
});
