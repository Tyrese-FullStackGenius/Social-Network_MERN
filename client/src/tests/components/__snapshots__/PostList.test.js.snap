// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render PostList 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PostList
    posts={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          Testing postlist
</div>,
        Array [],
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Testing postlist",
        },
        "ref": null,
        "rendered": "Testing postlist",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            Testing postlist
</div>,
          Array [],
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Testing postlist",
          },
          "ref": null,
          "rendered": "Testing postlist",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
